android/src/main/java/com/ioreactnativesecurestorage/IoReactNativeSecureStorageModule.kt

```diff
@@ -1,25 +1,245 @@
 package com.ioreactnativesecurestorage
 
+import com.facebook.react.bridge.Promise
 import com.facebook.react.bridge.ReactApplicationContext
 import com.facebook.react.bridge.ReactContextBaseJavaModule
 import com.facebook.react.bridge.ReactMethod
-import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.WritableArray
+import com.facebook.react.bridge.WritableMap
+import com.facebook.react.bridge.WritableNativeArray
+import com.facebook.react.bridge.WritableNativeMap
+import com.pagopa.securestorage.SecureStorage
+import java.io.File
+import java.nio.charset.StandardCharsets
+
 
 class IoReactNativeSecureStorageModule(reactContext: ReactApplicationContext) :
   ReactContextBaseJavaModule(reactContext) {
 
+  private var forceManualEncryption: Boolean = false
+  private val storageDir: File = File(reactContext.applicationContext.filesDir, "secure-storage")
+  private val testsObject = IoReactNativeSecureStorageTests(reactContext)
+
+  /**
+   * Lazily initialize the secure storage object only when the variable is called,
+   * otherwise it won't be created. Once created the same object will be used during
+   * its lifecycle.
+   */
+  private val secureStorage: SecureStorage? by lazy {
+    try {
+      val secureStorage = SecureStorage.Builder(reactContext.applicationContext, storageDir)
+        .setEnforceManualEncryption(forceManualEncryption)
+      secureStorage.build()
+    } catch (e: Exception) {
+      null
+    }
+  }
+
   override fun getName(): String {
     return NAME
   }
 
-  // Example method
-  // See https://reactnative.dev/docs/native-modules-android
+  /**
+   * @see [SecureStorage.Builder.setEnforceManualEncryption]
+   */
+  @ReactMethod
+  fun setEnforceManualEncryption(isEnforced: Boolean, promise: Promise) {
+    this.forceManualEncryption = isEnforced
+    promise.resolve(null)
+  }
+
```

Comment:
What about adding a function like that:

```kotlin
private fun performSecureAction(promise: Promise, action: SecureStorage.() -> Unit) {
  secureStorage?.run(action) ?: ModuleException.SECURE_STORE_NOT_INITIALIZED.reject(promise)
}
```

To write things like that, for example?

```kotlin
@ReactMethod
  fun get(key: String, promise: Promise) {
    Thread {
      try {
        performSecureAction(promise) {
          val value = get(key)
          value?.let {
            promise.resolve(it.toString(StandardCharsets.UTF_8))
          } ?: ModuleException.VALUE_NOT_FOUND.reject(promise)
        }
      } catch (e: Exception) {
        ModuleException.GET_FAILED.reject(
          promise, Pair(ERROR_USER_INFO_KEY, getExceptionMessageOrEmpty(e))
        )
      }
    }.start()
  }

```

Do you like it more? What do you think?

Reply:
Makes sense, thanks for the suggestion. Applied in df10b87ae2259ffa8c0e4bf65123f146e457325d.